### PR TITLE
Fix bug when user attempts to log in with expired token in Authorization header

### PR DESCRIPTION
**Pull request by nagem**
Ignore `Authorization` header flow when serving results to `/api/login`.

### Review Checklist

- Tests were added to cover all code changes
- Documentation was added / updated
- Code and tests follow standards in CONTRIBUTING.md

### CODE REVIEW

File: api/auth/authproviders.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def validate_code(self, code, **kwargs):
             'access_token': code,
             'uid': uid,
             'auth_type': self.auth_type,
-            'expires': datetime.datetime.utcnow() + datetime.timedelta(hours=1)
+            'expires': datetime.datetime.utcnow() + datetime.timedelta(days=14)
```
Comment (nagem) [inline api/auth/authproviders.py]: Since JWT has no refresh flow, I set this to be the default of our server that issues JWTs.

File: api/web/base.py [hunk 2]
```diff
@@ -184,6 +191,8 @@ def log_in(self):
 
         config.db.authtokens.insert_one(token_entry)
 
+        self.set_origin(False)
```
Comment (kofalt) [inline api/web/base.py]: What does this line do?

File: api/web/base.py [hunk 3]
```diff
@@ -173,6 +178,8 @@ def log_in(self):
         token_entry = auth_provider.validate_code(payload['code'], registration_code=registration_code)
         timestamp = datetime.datetime.utcnow()
 
+        self.uid = token_entry['uid']
```
Comment (kofalt) [inline api/web/base.py]: What was broken without this line?

File: api/web/base.py [hunk 4]
```diff
@@ -27,6 +27,11 @@ def __init__(self, request=None, response=None): # pylint: disable=super-init-no
         """Set uid, source_site, public_request, and superuser"""
         self.initialize(request, response)
 
+        # If user is attempting to log in through `/login`, ignore Auth here:
+        # In future updates, move login and logout handlers to class that overrides this init
+        if self.request.path == 'api/login':
+            return
```
Comment (kofalt) [inline api/web/base.py]: Ah. As an update to my "What was broken without this line?" comment, are these two pieces of code linked? Basically bypass legacy logic, and only `uid` was missing?

File: api/auth/authproviders.py (hunk 1 above)
Comment (kofalt) [inline api/auth/authproviders.py]: I'm concerned about this line. Won't we cache any JWT session for two weeks without verifying?
Why can't we respect the JWT expiry, if specified?

File: api/auth/authproviders.py [hunk 5]
```diff
@@ -128,7 +128,6 @@ def refresh_token(self, token):
 
         response = json.loads(r.content)
         return {
-            'refresh_token': response['refresh_token'],
```
Comment (kofalt) [inline api/auth/authproviders.py]: Why was this removed?

**Comment by nagem** [comment]
Tested changes with client that requests `access_type=offline` to ensure Google auth requests get a refresh token. Will merge when green.

File: api/auth/authproviders.py (hunk 5 above)
Comment (nagem) [inline api/auth/authproviders.py]: Google doesn't give us a refresh_token when requesting a new access_token like Wechat does.

File: api/web/base.py (hunk 4 above)
Comment (nagem) [inline api/web/base.py]: Legacy wouldn't be the right word. This logic is used to AuthX a request when the header contains auth information (session token/drone info/API key). For login/out endpoints, we should bypass this logic as we aren't trying to evaluate the authx via the headers.

File: api/auth/authproviders.py (hunk 1 above)
Comment (nagem) [inline api/auth/authproviders.py]: That is the JWT expiry for our server we are using. JWTs inherently can't be "refreshed". New work would need to be done to support a flow that "re-checks" rather than refreshes. That can be addressed in a separate PR. This change is to prevent users from being logged out every hour when their JWT token really lasts 14 days.

Comment (nagem) [inline api/auth/authproviders.py]: Google doesn't give us a refresh_token when requesting a new access_token like Wechat does.

Comment (kofalt) [inline api/auth/authproviders.py]: I'm all for leaving the refresh logic alone - my question was specifically about why we can't respect the JWT expiry instead of setting our own duration?

Comment (kofalt) [inline api/auth/authproviders.py]: Cool. Can you add either a comment talking about why this duration is set, or add a ticket to make the value configurable?

File: api/web/base.py (hunk 3 above)
Comment (nagem) [inline api/web/base.py]: In order to set the origin we would need to set self.uid

File: api/web/base.py (hunk 2 above)
Comment (nagem) [inline api/web/base.py]: Set the origin after the uid is known.

File: api/auth/authproviders.py (hunk 1 above)
Comment (nagem) [inline api/auth/authproviders.py]: I'll add a ticket to make this easier to understand/request that our JWT service gives us an expiry timestamp.
